ipa-core/src/protocol/ipa_prf/prf_sharding/feature_label_dot_product.rs

```diff
@@ -94,19 +98,29 @@ impl InputsRequiredFromPrevRow {
         )
         .await?;
 
-        let capped_attributed_feature_vector = select(
-            ctx.narrow(&Step::ComputedCappedFeatureVector),
-            record_id,
-            &capped_label,
-            &input_row.feature_vector,
-            &Replicated::<FV>::ZERO,
-        )
-        .await?;
+        let feature_contexts = (0..).map(|i| ctx.narrow(&Step::ComputedCappedFeatureVector(i)));
+        let capped_label_ref = &capped_label;
+        let capped_attributed_feature_vector = ctx
+            .parallel_join(zip(feature_contexts, &input_row.feature_vector).map(
+                |(c, feature)| async move {
+                    select(
+                        c,
+                        record_id,
+                        capped_label_ref,
+                        feature,
+                        &Replicated::<FV>::ZERO,
+                    )
+                    .await
+                },
+            ))
+            .await?;
```

Comment:
You may be able to simplify this with the stuff I added in https://github.com/private-attribution/ipa/pull/1046.